modules/tensor_mechanics/src/materials/RadialReturnCreepStressUpdateBase.C

```diff
@@ -0,0 +1,69 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "RadialReturnCreepStressUpdateBase.h"
+
+template <>
+InputParameters
+validParams<RadialReturnCreepStressUpdateBase>()
+{
+  InputParameters params = validParams<RadialReturnStressUpdate>();
+
+  params.addCoupledVar("temperature", 0.0, "Coupled temperature");
```

Comment:
Also in the stupid defaults department, temperature shouldn't have defaulted to 0...

Reply:
That's easily fixed by making it required. Just change this to `addRequiredCoupledVar`. If they weren't specifying it before, they'll get an error, and if they really wanted it to be zero, they can say `temperature=0` in the input file.

Reply:
Yeah okay, that'll change the _has_temp behavior which is protecting a divide by 0

Reply:
I think that can get yanked as well, and do a zero check on temperature instead. It's safer that way anyways in case the model returns a coupled zero temperature value.

Reply:
whoa wait a second, it's not even calculating the temperature exponential on the fly anyways!! It's only doing it at the beginning of the timestep!

Reply:
Defaulting temperature to zero seems okay too. It's just saying that creep strain is temperature independent?

Reply:
Most of the tests now assume temperature is 0.

Reply:
The rather, maybe a default of 0 is silly, but making it required doesn't seem the right way either.